src/ray/common/scheduling/cluster_resource_scheduler.cc

```diff
@@ -194,3 +271,226 @@ bool ClusterResourceScheduler::GetNodeResources(int64_t node_id,
 }
 
 int64_t ClusterResourceScheduler::NumNodes() { return nodes_.size(); }
+
+void ClusterResourceScheduler::ResourceMapToNodeResources(
+    const std::unordered_map<std::string, double> &resource_map_total,
+    const std::unordered_map<std::string, double> &resource_map_available,
+    NodeResources *node_resources) {
+  node_resources->capacities.resize(PredefinedResources_MAX);
+  for (size_t i = 0; i < PredefinedResources_MAX; i++) {
+    node_resources->capacities[i].total = node_resources->capacities[i].available = 0;
+  }
+
+  for (auto it = resource_map_total.begin(); it != resource_map_total.end(); ++it) {
+    ResourceCapacity resource_capacity;
+    resource_capacity.total = (int64_t)it->second;
+    auto it2 = resource_map_available.find(it->first);
+    if (it2 == resource_map_available.end()) {
+      resource_capacity.available = 0;
+    } else {
+      resource_capacity.available = (int64_t)it2->second;
+    }
+    if (it->first == "CPU") {
+      node_resources->capacities[CPU] = resource_capacity;
+    } else if (it->first == "GPU") {
+      node_resources->capacities[GPU] = resource_capacity;
+    } else if (it->first == "TPU") {
+      node_resources->capacities[TPU] = resource_capacity;
+    } else if (it->first == "memory") {
+      node_resources->capacities[MEM] = resource_capacity;
```

Comment:
Would be great to keep these string constants in an array instead of hard-coded here.

Reply:
done